### PR TITLE
Respect (current_tree_path parameter) on opening media gallery

### LINKED ISSUE #1362: The Directory filter is applied for non existing folder in Enhanced Media Gallery if Delete a folder from NOT Enhanced Media Gallery
### Preconditions (*)
1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Go to _Stores - Configuration - Advanced -System - Enhanced Media Gallery_ and select **Enabled - No**
enabled_no

### Steps to reproduce (*)
1. Go to **Content - Media Gallery**
2. **Create Folder**, name it "Unlicensed" for example 
3. Select the previously created folder from the folder tree, so that the _Directory_ filter is applied
4. Go to _Content - Pages_ click **Add New Page**
5. Expand _Content_ click **Show / Hide Editor** - **Insert Image...**
6. **Delete** the previously created folder
7. Go to **Content - Media Gallery** again

### Expected result (*)
The _Directory_ filter is not applied

### Actual result (*)
The _Directory_ filter is applied for the non-existing folder
directory_applied_deleted

### Note
- does NOT cause any error in dev console or issues with saving or uploading images

### PR DESCRIPTION
**Pull request by Nazar65**
### Description (*)

1. magento/adobe-stock-integration#1335: Requested default folder is not selected on opening media gallery (current_tree_path parameter) 
2.  magento/adobe-stock-integration#1362: The Directory filter is applied for non existing folder in Enhanced Media Gallery if Delete a folder from NOT Enhanced Media Gallery

### Manual testing scenarios (*)
1. Open media gallery from the category image field
2. Open media gallery from the WYSIWYG

### Expected result (*)
1. catalog/category directory is selected by default
2. WYSIWYG directory is selected by default

### CODE REVIEW

**Comment by Nazar65** [comment]
This PR has dependency to magento 2 repo Pull Request, i will provide link, soon.

**Comment by Nazar65** [comment]
https://github.com/magento/magento2/pull/28219

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 1]
```diff
@@ -96,10 +104,17 @@ define([
          * Verify directory filter on init event, select folder per directory filter state
          */
         checkChipFiltersState: function () {
-            if (!_.isUndefined(this.filterChips().filters.path) && this.filterChips().filters.path !== '') {
+            var currentFilterPath = this.filterChips().filters.path,
+                currentTreePath = Base64.decode(window.MediabrowserUtility.pathId.replace(/--|,,/, ''));
```
Comment (gabrieldagama) [inline MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: We probably need to perform the same change done on the magento 2 PR, decode the pathId using a similar method that PHP does.

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 2]
```diff
@@ -96,10 +104,17 @@ define([
...
+
+            if (!_.isUndefined(currentFilterPath) && currentFilterPath !== '' &&
```
Comment (gabrieldagama) [inline MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: I would extract this if statement to a function, just to make it easier to read and make the intended explicit.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @Nazar65, thanks for your pull request, please follow below some comments.

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js (hunk 1 above)
Comment (Nazar65) [reply MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: that is not required when we decode value

**Comment by Nazar65** [comment]
@magetno run all tests

**Comment by gabrieldagama** [comment]
Hi @Nazar65, also found an issue here, please follow the steps below to reproduce it:

- Go to Content -> Media -> Media Gallery (standalone media gallery)
- Open any folder (the filter will be applied)
- Reload the page

You should see the following error and the filter won't be applied:
```
directoryTree.js:108 Uncaught TypeError: Cannot read property 'pathId' of undefined
    at UiClass.checkChipFiltersState (directoryTree.js:108)
    at UiClass.<anonymous> (directoryTree.js:89)
    at HTMLDivElement.dispatch (jquery.js:5232)
    at HTMLDivElement.elemData.handle (jquery.js:4884)
    at Object.trigger (jquery.js:5136)
    at Object.jQuery.event.trigger (jquery-migrate.js:633)
    at jQuery.fn.init.triggerHandler (jquery.js:5872)
    at Object.__callback (jquery.jstree.js:247)
    at Object.loaded (jquery.jstree.js:549)
    at Object.$.jstree._fn.<computed> [as loaded] (jquery.jstree.js:244)```

Can you verify this, please?

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 3]
```diff
@@ -101,13 +109,56 @@ define([
          * Verify directory filter on init event, select folder per directory filter state
          */
         checkChipFiltersState: function () {
```
Comment (gabrieldagama) [inline MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js:111]: I think we can simplify this method, we are doing 3 times pretty much the same thing:
```
if (this.isFolderExistsInTree(currentFilterPath)) {
                    selectedId === currentFilterPath  || this.locateNode(currentFilterPath);
                } else {
                    this.selectStorageRoot();
                }
```

I believe something like this should work:
```
        /**
         * Verify directory filter on init event, select folder per directory filter state
         */
        checkChipFiltersState: function () {
            var currentFilterPath = this.filterChips().filters.path,
                isMediaBrowser = !_.isUndefined(window.MediabrowserUtility),
                currentTreePath;

            currentTreePath = (this.isFiltersApplied(currentFilterPath) || !isMediaBrowser) ? currentFilterPath :
                Base64.idDecode(window.MediabrowserUtility.pathId) ;

            if (this.isFolderExistsInTree(currentTreePath)) {
                this.locateNode(currentTreePath);
            } else {
                this.selectStorageRoot();
            }
        },
```

We just need to change the locateNode, to perform the validation as shown below:

```
        /**
         * Locate and higlight node in jstree by path id.
         *
         * @param {String} path
         */
        locateNode: function (path) {
            var selectedId =  $(this.directoryTreeSelector).jstree('get_selected').attr('id');

            if (path === selectedId) {
                return;
            }
            path = path.replace(/\//g, '\\/');
            $(this.directoryTreeSelector).jstree('open_node', '#' + path);
            $(this.directoryTreeSelector).jstree('select_node', '#' + path, true);

        }
```

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 4]
```diff
@@ -101,13 +109,56 @@ define([
...
+        isFolderExistsInTree: function (path) {
+            return $('#' + path.replace(/\//g, '\\/')).length === 1;
+        },
+
+        /**
+         * Check if need to select directory by filters state
+         *
+         * @param {String} currentFilterPath
+         */
+        isFiltersApllied: function (currentFilterPath) {
```
Comment (gabrieldagama) [inline MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]:
```suggestion
        isFiltersApplied: function (currentFilterPath) {
```

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 5]
```diff
@@ -101,13 +109,56 @@ define([
...
+
             }
         },
 
+        /**
+         * Verify if directory exists in folder tree
+         *
+         * @param {String} path
+         */
+        isFolderExistsInTree: function (path) {
```
Comment (gabrieldagama) [inline MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]:
```suggestion
        folderExistsInTree: function (path) {
```

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @Nazar65, thanks for your pull request, it looks good! Please follow some suggestions below.

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Review by gabrieldagama — DISMISSED** [review]

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed

The Directory filter is dropped together with the folder

**Comment by Nazar65** [comment]
@magento run Functional Tests B2B

**Pull request merged**